src/NServiceBus.AwsLambda.SQS.Analyzer/AwsLambdaDiagnostics.cs

```diff
@@ -0,0 +1,110 @@
+             title: "LimitMessageProcessing is not supported in AWS Lambda",
+             messageFormat: "Concurrency-related settings can be configured with the Lambda API.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Error,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor DefineCriticalErrorActionNotAllowed = new DiagnosticDescriptor(
+             id: DefineCriticalErrorActionNotAllowedId,
+             title: "DefineCriticalErrorAction is not supported in AWS Lambda",
+             messageFormat: "AWS Lambda endpoints do not control the application lifecycle and should not define behavior in the case of critical errors.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Error,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor SetDiagnosticsPathNotAllowed = new DiagnosticDescriptor(
+             id: SetDiagnosticsPathNotAllowedId,
+             title: "SetDiagnosticsPath is not supported in AWS Lambda",
+             messageFormat: "AWS Lambda endpoints should not write diagnostics to the local file system. Use CustomDiagnosticsWriter to write diagnostics to another location.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Error,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor MakeInstanceUniquelyAddressableNotAllowed = new DiagnosticDescriptor(
+             id: MakeInstanceUniquelyAddressableNotAllowedId,
+             title: "MakeInstanceUniquelyAddressable is not supported in AWS Lambda",
+             messageFormat: "AWS Lambda endpoints have unpredictable lifecycles and should not be uniquely addressable.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Error,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor UseTransportNotAllowed = new DiagnosticDescriptor(
+             id: UseTransportNotAllowedId,
+             title: "UseTransport is not supported in AWS Lambda",
+             messageFormat: "The package configures Amazon SQS transport by default. Use AwsLambdaSQSEndpointConfiguration.Transport to access the transport configuration.",
+             category: DiagnosticCategory,
+             defaultSeverity: DiagnosticSeverity.Warning,
+             isEnabledByDefault: true
+            );
+
+        internal static readonly DiagnosticDescriptor OverrideLocalAddressNotAllowed = new DiagnosticDescriptor(
+             id: OverrideLocalAddressNotAllowedId,
+             title: "OverrideLocalAddress is not supported in AWS Lambda",
+             messageFormat: "The NServiceBus endpoint address in AWS Lambda is determined by the ServiceBusTrigger attribute.",
```

Comment:
I think this language is correct based on [the sample in docs](https://github.com/Particular/docs.particular.net/blob/master/samples/hosting/aws-lambda-sqs/SQSLambda_1/ServerlessEndpoint/serverless.template#L54-L67). @timbussmann can you verify?

```suggestion
             messageFormat: "The NServiceBus endpoint address in AWS Lambda is determined by the serverless template.",
```

Reply:
@mikeminutillo I don't think it's determined by the template. The CloudFormation template in the sample is used to setup the necessary infrastructure (since we can't use installers in serverless environments). I'm not too familiar with AWS Lambda but I assume the endpoint address is determined by the `AwsLambdaSQSEndpointConfiguration` configuration parameter. The current text definitely seems to be off though.

Reply:
Looks like it is the [event source mapping](https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventsourcemapping.html)

Reply:
Updated here https://github.com/Particular/NServiceBus.AwsLambda.Sqs/pull/306